cmd/thanos/compact.go

```diff
@@ -623,9 +638,13 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	cmd.Flag("compact.enable-vertical-compaction", "Experimental. When set to true, compactor will allow overlaps and perform **irreversible** vertical compaction. See https://thanos.io/tip/components/compact.md/#vertical-compactions to read more."+
 		"Please note that this uses a NAIVE algorithm for merging (no smart replica deduplication, just chaining samples together)."+
-		"NOTE: This flag is ignored and (enabled) when --deduplication.replica-label flag is set.").
+		"NOTE: This flag is ignored and (enabled) when --deduplication.replica-label or --compact.enable-dedup flag is set.").
```

Comment:
Why we need 3rd flag for enabling dedup? (:

Reply:
If we add a dedup algorithm enum flag then this can be removed for sure.